client/app/bundles/course/assessment/submission/pages/SubmissionEditIndex/index.jsx

```diff
@@ -54,10 +64,42 @@ class VisibleSubmissionEditIndex extends Component {
     dispatch(autogradeSubmission(params.submissionId));
   }
 
+  /**
+   * Return true if it is ready to submit
+   */
+  validateSubmit = (showErrorNotification = true) => {
+    const { dispatch, form } = this.props;
+    const answers = Object.values(form.values);
+
+    // touch all the values
+    answers.forEach((answer = {}) => {
+      const answerId = answer.id;
+      Object.keys(answer).forEach((key) => {
+        dispatch(touch(formNames.SUBMISSION, `${answerId}.${key}`));
+      });
+    });
+
+    let readyToSubmit = true;
+    Object.values(form.syncErrors || {}).forEach((answerError) => {
+      const noError = Object.keys(answerError).length === 0;
+      if (!noError) {
+        readyToSubmit = false;
+      }
+    });
+
+    if (!readyToSubmit && showErrorNotification) {
+      setNotification(translations.submitError)(dispatch);
```

Comment:
Would be better if you can take the notification logic out and keep the method single responsibility. Now this method returns a value + set the notification.